telethon/utils.py

```diff
@@ -1101,104 +1103,85 @@ def resolve_bot_file_id(file_id):
                 duration=0,
-                voice=file_type == 3
+                voice=file.type_id == FileId.TYPE_VOICE
             ))
-        elif file_type == 4 or file_type == 13:
+        elif file.type_id == FileId.TYPE_VIDEO or file.type_id == FileId.TYPE_VIDEO_NOTE:
             attributes.append(types.DocumentAttributeVideo(
                 duration=0,
                 w=0,
                 h=0,
-                round_message=file_type == 13
+                round_message=file.type_id == FileId.TYPE_VIDEO_NOTE
             ))
-        # elif file_type == 5:  # other, cannot know which
-        elif file_type == 8:
+        # elif file.type_id == FieldId.TYPE_DOCUMENT:  # other, cannot know which --> no attribute
+        elif file.type_id == FileId.TYPE_STICKER:
             attributes.append(types.DocumentAttributeSticker(
                 alt='',
                 stickerset=types.InputStickerSetEmpty()
             ))
-        elif file_type == 10:
+        elif file.type_id == FileId.TYPE_ANIMATION:
             attributes.append(types.DocumentAttributeAnimated())
+        # end if
 
         return types.Document(
-            id=media_id,
-            access_hash=access_hash,
+            id=file.id,
+            access_hash=file.access_hash,
             date=None,
             mime_type='',
             size=0,
             thumbs=None,
-            dc_id=dc_id,
+            dc_id=file.dc_id,
             attributes=attributes,
-            file_reference=b''
+            file_reference=file.file_reference if file.has_reference else b''
         )
-    elif (version == 2 and len(data) == 44) or (version == 4 and len(data) in (49, 77)):
-        if version == 2:
-            (file_type, dc_id, media_id, access_hash,
-                volume_id, secret, local_id) = struct.unpack('<iiqqqqi', data)
-        # else version == 4:
-        elif len(data) == 49:
-            # TODO Figure out what the extra five bytes mean
-            (file_type, dc_id, media_id, access_hash,
-                volume_id, secret, local_id, _) = struct.unpack('<iiqqqqi5s', data)
-        elif len(data) == 77:
-            # See #1613.
-            (file_type, dc_id, _, media_id, access_hash, volume_id, _, local_id, _) = struct.unpack('<ii28sqqq12sib', data)
-        else:
-            return None
+    elif not isinstance(file, PhotoFileId):  # e.g. WebLocationFileId
+        return None
+    # end if
 
-        if not (1 <= dc_id <= 5):
-            return None
+    # okey, so we're dealing with a photo.
+    assert isinstance(file, PhotoFileId)
+    photo_size = 's' if file.id or file.access_hash else 'x'
 
-        # Thumbnails (small) always have ID 0; otherwise size 'x'
-        photo_size = 's' if media_id or access_hash else 'x'
-        return types.Photo(
-            id=media_id,
-            access_hash=access_hash,
-            file_reference=b'',
-            date=None,
-            sizes=[types.PhotoSize(
+    foto_size = file.photosize
```

Comment:
Let's make this a bit more readable and give this an actual English name (and change the uses below too):

```python
    photo_type = 's' if file.id or file.access_hash else 'x' 
    photo_size = file.photosize
    if isinstance(photo_size, (
        PhotoFileId.PhotosizeSourceLegacy,
        PhotoFileId.PhotosizeSourceThumbnail,
        PhotoFileId.PhotosizeSourceDialogPhotoSmall,
        PhotoFileId.PhotosizeSourceDialogPhotoBig,
        PhotoFileId.PhotosizeSourceStickersetThumbnail,
    )):
```